### PR TITLE
Refactor cpp code in Gold/MST

### PR DESCRIPTION
**Pull request by ChuangSheep**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: content/4_Gold/MST.mdx [hunk 1]
```diff
@@ -144,64 +144,61 @@ no edges are counted twice, we cannot "accidentally" count $n-1$ edges.
...
-		x = get(x), y = get(y); if (x == y) return 0;
-		if (e[x] > e[y]) swap(x,y);
-		e[x] += e[y]; e[y] = x; return 1;
+	bool unite(int x, int y) { 
+		x = get(x), y = get(y); 
+		if (x == y) {
+			return false;
+		}
+		if (e[x] > e[y]) {
+			swap(x,y);
```
Comment (SansPapyrus683) [inline content/4_Gold/MST.mdx:162]: spaces

File: content/4_Gold/MST.mdx [hunk 2]
```diff
@@ -144,64 +144,61 @@ no edges are counted twice, we cannot "accidentally" count $n-1$ edges.
...
 
-int cnt=0;
-template<class T> T kruskal(int N, vector<pair<T,pi>> ed) {
-	sort(all(ed));
-	T ans = 0; DSU D; D.init(N+1); // edges that unite are in MST
-	trav(a,ed)
-	if (D.unite(a.s.f,a.s.s)) {
-		ans += a.f;
-		cnt++;
+template<class T> T kruskal(int N, vector<pair<T, pair<int, int>>> edges) {
```
Comment (SansPapyrus683) [inline content/4_Gold/MST.mdx:170]: why the generic?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
yeaaaah

File: content/4_Gold/MST.mdx (hunk 2 above)
Comment (ChuangSheep) [reply content/4_Gold/MST.mdx:170]: it was generic and I didnt change it, don't think this is too big an issue

File: content/4_Gold/MST.mdx [hunk 3]
```diff
@@ -144,64 +144,61 @@ no edges are counted twice, we cannot "accidentally" count $n-1$ edges.
 <CPPSection>
 
 ```cpp
-#include <iostream>
-#include <vector>
-#include <algorithm>
+#include <bits/stdc++.h>
 using namespace std;
```
Comment (SansPapyrus683) [inline content/4_Gold/MST.mdx:148]: can you revert this?

**Review by SansPapyrus683 — APPROVED** [review]
should be good enough?

File: content/4_Gold/MST.mdx (hunk 3 above)
Comment (ChuangSheep) [reply content/4_Gold/MST.mdx:148]: tbh i dont think it's more clear to have 5 lines of include statements, but sure

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**